servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/ConnectionObserverInitializer.java

```diff
@@ -75,15 +78,37 @@ public ConnectionObserverInitializer(final ConnectionObserver observer,
      * {@link Channel} to report {@link ConnectionObserver#onTransportHandshakeComplete(ConnectionInfo)}
      * @param handshakeOnActive {@code true} if the observed connection is secure
      * @param client {@code true} if this initializer is used on the client-side
+     * @deprecated Use {@link #ConnectionObserverInitializer(ConnectionObserver, Function, boolean, boolean, SslConfig)}
+     * instead
      */
+    @Deprecated // FIXME: 0.43 - remove deprecated ctor
     public ConnectionObserverInitializer(final ConnectionObserver observer,
                                          final Function<Channel, ConnectionInfo> connectionInfoFactory,
                                          final boolean handshakeOnActive,
                                          final boolean client) {
+        this(observer, connectionInfoFactory, handshakeOnActive, client, null);
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param observer {@link ConnectionObserver} to report network events
+     * @param connectionInfoFactory {@link Function} that creates {@link ConnectionInfo} from the provided
+     * {@link Channel} to report {@link ConnectionObserver#onTransportHandshakeComplete(ConnectionInfo)}
+     * @param handshakeOnActive {@code true} if the observed connection is secure
+     * @param client {@code true} if this initializer is used on the client-side
+     * @param sslConfig the {@link SslConfig} to supply to the observer on handshake.
+     */
+    public ConnectionObserverInitializer(final ConnectionObserver observer,
+                                         final Function<Channel, ConnectionInfo> connectionInfoFactory,
+                                         final boolean handshakeOnActive,
```

Comment:
`handshakeOnActive` is effectively a `config.sslConfig() != null` check. Consider using `sslConfig` instead of yet another ctor argument. For `TcpClientChannelInitializer` we will pass `sslConfig` only if `!deferSslHandler`.

Reply:
In the client case it is `sslConfig != null && !deferSslHandler`, an in the deprecation overrides we can only set null so it will always be false. The second case might be ok, but how would we deal with the deferSslHandler check unless adding another parameter too?